### ISSUE TITLE
Environments do not match available clouds

### ISSUE THREAD

**Issue by dsacora** [Azure/go-autorest#624]
From https://github.com/GoogleCloudPlatform/terraformer/issues/654

Using terraformer which uses in turn go-autorest it fails when using the "az login" authentication. 

The error from this is: 

An Azure Environment with name "" was not found: autorest/azure: There is no cloud environment matching the name "AZURECLOUD"

Tracing that error back we come to this file:
https://github.com/Azure/go-autorest/blob/master/autorest/azure/environments.go

Within that we have this block:

var environments = map[string]Environment{
"AZURECHINACLOUD": ChinaCloud,
"AZUREGERMANCLOUD": GermanCloud,
"AZUREPUBLICCLOUD": PublicCloud,
"AZUREUSGOVERNMENTCLOUD": USGovernmentCloud,
}

However if we look at what is available we get this:
az cloud list -o table
IsActive Name Profile

True AzureCloud latest
False AzureChinaCloud latest
False AzureUSGovernment latest
False AzureGermanCloud latest

I believe here that AzureChinaCloud and AzureGermanCloud would match, the others do not.

**Comment by jhendrixMSFT** [comment]
The name they should be using is `AzurePublicCloud`, not `AzureCloud`.  While I agree it's unfortunate that the names don't match, it's been like this for years so we can't change it now without breaking existing customers.  The terraformer app will need to perform a translation of these cloud names.

**Comment by cesarca** [comment]
Something new?

**Comment by yves-vogl** [comment]
@jhendrixMSFT [jhendrixMSFT](https://github.com/jhendrixMSFT) I know that the right way would be a pull request. But first let me know if the following proposal would be ok for you:

1. Introduce *AzureCloud* as alias for *AzurePublicCloud*
2. Mark *AzurePublicCloud* as deprecated and log a corresponding message

If that's ok for you I'll create a PR.

**Comment by jhendrixMSFT** [comment]
@yves-vogl while what you propose makes sense, there is a concern about this long-term.  Our new module, `azidentity`, which will replace `go-autorest` for authentication, also calls this [`AzurePublicCloud`](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azidentity#AuthorityHost).  This name also aligns with our SDKs in other languages.

Can I get more info on the scenario here?  Maybe there's a different way to solve this using existing constructs.

**Comment by jhendrixMSFT** [comment]
CC @chlowell

**Comment by dsacora** [comment]
> @yves-vogl while what you propose makes sense, there is a concern about this long-term. Our new module, `azidentity`, which will replace `go-autorest` for authentication, also calls this [`AzurePublicCloud`](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azidentity#AuthorityHost). This name also aligns with our SDKs in other languages.
> 
> Can I get more info on the scenario here? Maybe there's a different way to solve this using existing constructs.

Being honest I'd given up on this due to the "don't care if it's broken, won't fix" attitude (and dropped the entire idea of using terraformer because of it) but seeing as there is activity again... 

Both the Azure CLI and Powershell refer to it as AZURECLOUD. It presumably has been changed at some point in the past by someone else at Microsoft *OR* this is not universal and different countries/languages see different things.

I have no idea if AZUREPUBLICCLOUD is actually used anymore (or maybe it is used in other countries) but it certainly is not listed in the providers in the UK. 

IMO AZUREPUBLICCLOUD should be an alias for AZURECLOUD (or vv). 

I will repeat it again, this is the list of providers. AzurePublicCloud is not one of them.  

```
PS /home/microsoft> az cloud list -o table
IsActive    Name               Profile
----------  -----------------  ---------
True        AzureCloud         latest
False       AzureChinaCloud    latest
False       AzureUSGovernment  latest
False       AzureGermanCloud   latest
```

**Comment by serefacet** [comment]
I agree with @dsacora. I'm not seeing any downside of adding a new alias "AzureCloud" to the environment. Please correct me if I'm wrong.

**Comment by 24Phani** [comment]
2022/03/27 15:21:05 Testing if Service Principal / Client Certificate is applicable for Authentication..
2022/03/27 15:21:05 Testing if Multi Tenant Service Principal / Client Secret is applicable for Authentication..
2022/03/27 15:21:05 Testing if Service Principal / Client Secret is applicable for Authentication..
2022/03/27 15:21:05 Testing if Managed Service Identity is applicable for Authentication..
2022/03/27 15:21:05 Testing if Obtaining a token from the Azure CLI is applicable for Authentication..
2022/03/27 15:21:05 Using Obtaining a token from the Azure CLI for Authentication
2022/03/27 15:21:05 An Azure Environment with name "" was not found: autorest/azure: There is no cloud environment matching the name "AZURECLOUD"

anybody have a fix for this issue, please let me know.

**Comment by pfgarvey** [comment]
So, in other words, terraformer is completely useless for azure public cloud (pick your alias)? I spent two hours investigating this and now this is the stopping point???

**Comment by jhendrixMSFT** [comment]
Fixed in `autorest@v0.11.27`
